app/controllers/component-tree.js

```diff
@@ -1,314 +1,358 @@
+  itemHeight = 22;
+
+  @controller application;
+  @service port;
+
+  @tracked query = '';
+  @tracked isInspecting = false;
+  @tracked renderItems = [];
+
+  _store = Object.create(null);
+
+  set renderTree(renderTree) {
+    let { _store } = this;
+
+    let store = Object.create(null);
+    let renderItems = [];
+
+    let flatten = (parent, renderNode) => {
+      if (isInternalRenderNode(renderNode)) {
+        renderNode.children.forEach(node => flatten(parent, node));
       } else {
-        expandedStateCache[getIdFromObj(viewItem)] = viewItem.expanded;
+        let item = _store[renderNode.id];
+
+        if (item === undefined) {
+          item = new RenderItem(this, parent, renderNode);
+        } else {
+          item.renderNode = renderNode;
+        }
+
+        store[renderNode.id] = item;
+
+        renderItems.push(item);
+
+        renderNode.children.forEach(node => flatten(item, node));
       }
-    });
+    };
+
+    renderTree.forEach(node => flatten(null, node));
 
-    return viewArray;
-  }),
+    this._store = store;
+
+    this.renderItems = renderItems;
+  }
+
+  findItem(id) {
+    return this._store[id];
+  }
 
-  viewArray: computed('viewTree', 'searchValue', function() {
-    let tree = this.viewTree;
-    if (!tree) {
-      return [];
+  get currentItem() {
+    if (this.previewing) {
+      return this.findItem(this.previewing);
+    } else if (this.pinned) {
+      return this.findItem(this.pinned);
+    } else {
+      return undefined;
     }
-    return flattenSearchTree(this.searchValue, tree, null, 0, false, []);
-  }),
-
-  expandedStateCache: null, //set on init
-
-  init() {
-    this._super(...arguments);
-    this.set('expandedStateCache', {});
-  },
-
-  /**
-   * Expands the component tree so that entry for the given view will
-   * be shown.  Recursively expands the entry's parents up to the root.
-   * @param {*} objectId The id of the ember view to show
-   */
-  expandToNode(objectId) {
-    let node = this.filteredArray.find(item => item.get('id') === objectId);
-    if (node) {
-      node.expandParents();
+  }
+
+  get matchingItems() {
+    let { renderItems, query } = this;
+
+    if (query) {
+      let match = item => searchMatch(item.name, query) || item.childItems.some(match);
+      renderItems = renderItems.filter(match);
     }
-  },
-
-  /**
-   * This method is basically a trick to get the `{{vertical-collection}}` in the vicinity
-   * of the item that's been selected.  We can't directly scroll to the element but we
-   * can guess at how far down the list the item is. Then we can manually set the scrollTop
-   * of the virtual scroll.
-   */
-  scrollTreeToItem(objectId) {
-    let selectedItemIndex = this.displayedList.findIndex(item => item.view.objectId === objectId);
-
-    if (!selectedItemIndex) {
+
+    return renderItems;
+  }
+
+  get visibleItems() {
+    return this.matchingItems.filter(item => item.isVisible);
+  }
+
+  @tracked _pinned = undefined;
```

Comment:
Not a giant issue, but I would have expected all of the tracked properties to be at the beginning of the class (essentially together).